src/games/behavmixed.imp

```diff
@@ -643,130 +589,119 @@ T MixedBehaviorProfile<T>::DiffNodeValue(const GameNode &p_node,
 //             MixedBehaviorProfile<T>: Cached profile information
 //========================================================================
 
+
+// compute realization probabilities for nodes and isets.
+template <class T>
+void MixedBehaviorProfile<T>::ComputePass1_realizProbs(const GameNode &node) const
+{
+  if (node->GetParent()) {
+    map_realizProbs[node] = map_realizProbs[(node->GetParent())] * GetActionProb(node->GetPriorAction());
+  }
+  else {
+    map_realizProbs[node] = (T) 1;
+  }
+
+  if (node->GetInfoset()) {
+    for (auto childNode: node->GetChildren()) {
+      ComputePass1_realizProbs(childNode);
+    }
+  }
+}
+
 template <class T>
-void MixedBehaviorProfile<T>::ComputeSolutionDataPass2(const GameNode &node) const
+void MixedBehaviorProfile<T>::ComputePass2_beliefs_nodeValues_actionValues(const GameNode &node) const
 {
   if (node->GetOutcome()) {
     GameOutcome outcome = node->GetOutcome();
-    for (int pl = 1; pl <= m_support.GetGame()->NumPlayers(); pl++) {
-      m_nodeValues(node->GetNumber(), pl) += static_cast<T>(outcome->GetPayoff(pl));
+    for (auto player: m_support.GetGame()->GetPlayers()) {
+      map_nodeValues[node][player] += static_cast<T>(outcome->GetPayoff(player));
     }
   }
 
   GameInfoset iset = node->GetInfoset();
 
   if (iset) {
```

Comment:
It would make the code more readable just to check if it's a terminal node and return, rather than having a long nested `if` conditional.